x/ibc/20-transfer/types/msgs.go

```diff
@@ -12,6 +12,7 @@ import (
 type MsgTransfer struct {
 	SourcePort    string         `json:"source_port" yaml:"source_port"`       // the port on which the packet will be sent
 	SourceChannel string         `json:"source_channel" yaml:"source_channel"` // the channel by which the packet will be sent
+	DestHeight    uint64         `json:"dest_height" yaml:"dest_height"`       // the current height of the destination chain
```

Comment:
Chose to add `DestHeight` as a field user must pass in, rather than trying to retrieve this from state

Trying to retrieve this from state will require bubbling up this information from the `ClientKeeper` which is the only keeper that has information on the `LatestHeight` of the destination chain. Since this is pretty far down in the abstraction barrier with respect to ICS20, not doing this for now

**But** if we're asking user to pass this info in, what is the point of passing in `DestHeight` and then adding in a timout. Instead we can just ask users to pass in a `TimeoutHeight` directly

Reply:
how would users know the destination chain current height? seems like it'd require extra queries from clients to retrieve this

Reply:
the problem with this approach is that the `destHeight` is not validated either. tbh I prefer the internal query since provides a better UX

Reply:
It's not really possible to validate it in a complete fashion, since the destination height is a property of the other chain. The most we could do is fetch the latest height of the local client, but it might not have been updated. The user themselves should be able to query the destination chain and figure out what the latest height is.

Reply:
Can I get a response regarding:

> But if we're asking user to pass this info in, what is the point of passing in DestHeight and then adding in a timout. Instead we can just ask users to pass in a TimeoutHeight directly

Reply:
Ah - passing in a timeout height directly is fine, yes.